src/infrastructure/logging.py

```
@@ -0,0 +1,153 @@
+        if record.exc_info:
+            exc_type, exc_value, exc_traceback = record.exc_info
+            record.error_type = exc_type.__name__ if exc_type else None
+            record.error_message = str(exc_value) if exc_value else None
+            record.stacktrace = traceback.format_exception(exc_type, exc_value, exc_traceback)
+        return True
+
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    def add_fields(self, log_record, record, message_dict):
+        super().add_fields(log_record, record, message_dict)
+        log_record['timestamp'] = datetime.utcnow().isoformat()
```

Ruff (0.12.2)

`datetime.datetime.utcnow()` used

(DTZ003)